src/main/java/de/chojo/repbot/analyzer/ContextResolver.java

```diff
@@ -18,29 +17,27 @@
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
 import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Collection;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Objects;
-import java.util.Set;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 import static org.slf4j.LoggerFactory.getLogger;
 
 public class ContextResolver {
     private static final Logger log = getLogger(ContextResolver.class);
     private final Voice voiceData;
     private final Configuration configuration;
-    private final Cache<Long, Set<Long>> messageContextCache = CacheBuilder.newBuilder()
+    private final Cache<Long, MessageContext> messageContextCache = CacheBuilder.newBuilder()
             .expireAfterAccess(5, TimeUnit.SECONDS)
             .expireAfterWrite(10, TimeUnit.SECONDS)
             .maximumSize(100000)
             .build();
 
-    private final Cache<Long, Set<Long>> voiceContextCache = CacheBuilder.newBuilder()
+    private final Cache<Long, MessageContext> voiceContextCache = CacheBuilder.newBuilder()
```

Comment:
its the context for a mesage based on the voice chat sharings of a user. could rename it to voiceMessageContextCache but I wanted to avoid overly long names

Reply:
This was more like just a note, not really a change request :smile: